modules/Graphs.tla

```diff
@@ -61,11 +61,18 @@ IsStronglyConnected(G) ==
 -----------------------------------------------------------------------------
 IsTreeWithRoot(G, r) ==
   /\ IsDirectedGraph(G)
-  /\ \A e \in G.edge : /\ e[1] # r
-                       /\ \A f \in G.edge : (e[1] = f[1]) => (e = f)
-  /\ \A n \in G.node : AreConnectedIn(n, r, G)
+  /\ (Cardinality(G.node) = 1 => G.node = {r})
+     \* Handles single-node tree case
+  /\ \A n \in G.node \ {r} : \E! f \in G.edge : f[2] = n
+     \* Every node except the root has exactly one incoming edge
+  /\ \A n \in G.node \ {r} : AreConnectedIn(r, n, G)
+     \* Every node other than the root is reachable from the root
+  /\ \A n \in G.node \ {r} : ~\E path \in SimplePath(G) : path[1] = n /\ path[Len(path)] = n

```

Comment:
\* Every node except the root has exactly one incoming edge
\* Every node other than the root is reachable from the root

What about a forest?

Reply:
you mean I should change the wording to `forest` or do you think I missed something?

Reply:
Never mind, it appears as if the original definition is also false for a forest.  It is probably a good idea to add tests that demonstrate the effects of your changes.